shotover-proxy/src/transforms/protect/crypto.rs

```diff
@@ -1,9 +1,10 @@
 use crate::message::MessageValue;
 use crate::transforms::protect::key_management::KeyManager;
 use anyhow::{anyhow, bail, Result};
+use chacha20poly1305::aead::{Aead, NewAead};
+use chacha20poly1305::{ChaCha20Poly1305, Key, Nonce};
+use rand::RngCore;
 use serde::{Deserialize, Serialize};
-use sodiumoxide::crypto::secretbox;
-use sodiumoxide::crypto::secretbox::Nonce;
 use sqlparser::ast::Value as SQLValue;
 
 // A protected value meets the following properties:
```

Comment:
This comment still talks about libsodium.
Also does our new library take care of "all padding, copying and timing issues associated with crypto" as well?

Reply:
I'm not sure whether padding refers to [this type of padding](https://datatracker.ietf.org/doc/html/rfc7539#:~:text=padding2%20%2D%2D%20the%20padding) or this [type](https://en.wikipedia.org/wiki/Padding_(cryptography)). Also unsure what copying refers to. I would assume that timing refers to the nonce which is handled.

@benbromhead ?

Reply:
Though it appears padding is handled with chacha20: https://loup-vaillant.fr/tutorials/chacha20-design

Reply:
Padding is handled by chacha20.
The AEAD spec rfc 7539 claims the design is supposed to lend itself to a constant time implementation such that timing attacks are minimised. At this point I can only assume and rely on audits such as https://github.com/RustCrypto/AEADs/issues/426 that the implementation is kind of sound. 

At this point in time, my priority is to move to a maintained / popular crypto library. 

For what its worth secretbox in sodium oxide is just a re-export of xsalsa20poly1305 and the libsodium implementation is the actual AEAD implementation https://github.com/jedisct1/libsodium/blob/master/src/libsodium/crypto_secretbox/xsalsa20poly1305/secretbox_xsalsa20poly1305.c 

I'm happy that with this change we are getting the same thing we _thought_ we were getting previously. Ideally when we need to use this in anger, we will get an independent audit of our use (reading the RFC, how we generate our nonce appears to be the main area we can trip up badly).